framework/src/actions/AddRelationshipManager.C

```diff
@@ -26,13 +26,29 @@ AddRelationshipManager::AddRelationshipManager(InputParameters params) : Action(
 void
 AddRelationshipManager::act()
 {
-  Moose::RelationshipManagerType rm_type =
-      (_current_task == "add_geometric_rm" ? Moose::RelationshipManagerType::GEOMETRIC
-                                           : Moose::RelationshipManagerType::ALGEBRAIC);
-
-  const auto & all_action_ptrs = _awh.allActionBlocks();
-  for (const auto & action_ptr : all_action_ptrs)
-    action_ptr->addRelationshipManagers(rm_type);
-
-  _app.attachRelationshipManagers(rm_type);
+  if (_current_task == "attach_geometric_rm")
+  {
+    // Add any more relationship managers that are needed
+    const auto & all_action_ptrs = _awh.allActionBlocks();
+    for (const auto & action_ptr : all_action_ptrs)
+      action_ptr->addRelationshipManagers(Moose::RelationshipManagerType::GEOMETRIC);
+
+    _app.attachRelationshipManagers(Moose::RelationshipManagerType::GEOMETRIC);
+  }
+
+  if (_current_task == "attach_algebraic_rm")
+  {
+    const auto & all_action_ptrs = _awh.allActionBlocks();
+    for (const auto & action_ptr : all_action_ptrs)
+      action_ptr->addRelationshipManagers(Moose::RelationshipManagerType::ALGEBRAIC);
+
+    _app.attachRelationshipManagers(Moose::RelationshipManagerType::ALGEBRAIC);
```

Comment:
Funny - this manual unroll created some code duplication. I think I liked it the other way better.